### PR TITLE
Homework4: Maria Filiptsova

### PR DESCRIPTION
**Pull request by MariaFiliptsova**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/Homework4/Cities.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Odessa
+Moscow
+Novgorod
+
+Выходные данные
+Ukraine
+Russia
+Russia
+"""
+N = int(input('Enter number of countries\n'))
```
Comment (belovmd) [inline src/Homework4/Cities.py:32]: please put empty line between task description and task

File: src/Homework4/Languages.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+N = int(input('Enter number of students\n'))
+all_languages = []
+spoken_by_everyone = set()
+for i in range(N):
+    M = int(input('Number of spoken languages\n'))
+    for j in range(M):
+        spoken_language = input('What language\n')
+        if spoken_language not in all_languages:
+            all_languages.append(spoken_language)
+        else:
```
Comment (belovmd) [inline src/Homework4/Languages.py]: redundant code

File: src/Homework4/Languages.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            pass
+# Пока не знаю как выести языки, которые знают все школьники.
```
Comment (belovmd) [inline src/Homework4/Languages.py]: missed second part

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework4.md):
- 17
- 18
- 20
- 28
- 29

File: src/Homework4/Cities.py (hunk 1 above)
Comment (MariaFiliptsova) [reply src/Homework4/Cities.py:32]: Done

File: src/Homework4/Languages.py (hunk 2 above)
Comment (MariaFiliptsova) [reply src/Homework4/Languages.py]: Rewritten.

File: src/Homework4/Languages.py (hunk 3 above)
Comment (MariaFiliptsova) [reply src/Homework4/Languages.py]: Added.

File: src/Homework4/Cities.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+countries_with_cities = {}
+repeated_cities = {}
+for i in range(N):
+    country, *cities = input('Enter country and its cities\n').split()
+    for city in cities:
+        if city not in countries_with_cities:
+            countries_with_cities.update({city: country})
+        else:
+            repeated_cities.update({city: country})
```
Comment (belovmd) [inline src/Homework4/Cities.py:41]: can be solved with next dict: {city: [country, country]}

File: src/Homework4/Lists.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+"""Даны два списка чисел.
+Посчитайте, сколько различных чисел содержится
+одновременно как в первом списке, так и во втором.
+"""
+
+# Решение 1
+# spisok1 = [1, 2, 3, 4]
+# spisok2 = [3, 4, 5, 6, 7, 1]
+# spisok3 = [chislo for chislo in spisok1 if chislo in spisok2]
```
Comment (belovmd) [inline src/Homework4/Lists.py:9]: redundant comments in code

File: src/Homework4/Slova.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import string
+
+"""Во входной строке записан текст.
+Словом считается последовательность непробельных символов идущих подряд,
+слова разделены одним или большим числом пробелов или символами конца строки.
+Определите, сколько различных слов содержится в этом тексте.
+"""
+
+# я не согласна, что не надо удалять знаки препинания, потому что,
+# если мы это не сделаем 'Пьер' и 'Пьер,' будут различными словами,
```
Comment (belovmd) [inline src/Homework4/Slova.py:10]: According to task description -  word is a sequence of non-space characters. Yes, it means that Car and car, are different words.

File: src/Homework4/NatChisla.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
+"""Даны два натуральных числа.
+Вычислите их наибольший общий делитель при помощи алгоритма Евклида
+(мы не знаем функции и рекурсию).
+"""
+
+# num1 = 10
+# num2 = 8
+# while num1 > 0 and num2 > 0:
+#     if num1 > num2:
+#         num1 = num1 % num2
```
Comment (belovmd) [inline src/Homework4/NatChisla.py:10]: trash comments

**Review by belovmd — COMMENTED** [review]
8

**Pull request closed without merging**